server/aesdsocket.c

```diff
@@ -0,0 +1,331 @@
+bool graceful_exit_handler = false;
+
+//signal handler for SIGINT and SIGTERM
+void signal_handler(int sig) {
+  if (sig == SIGINT || sig == SIGTERM) {
+    syslog(LOG_INFO, "Caught signal. Exiting...");
+    graceful_exit_handler = true;
+    exit(0);
+  }
+}
+
+void managesocket(int socket_t) {
+  int listen_ret = listen(socket_t, 10);
+  if (listen_ret == -1) { //keep socket size 10 to listen for 10 incoming connections in queue
+    syslog(LOG_ERR, "Error occured while listening to socket = %s. Exiting...", strerror(errno));
+    exit(-1);
+  }
+
+  syslog(LOG_INFO, "Listening on port 9000");
+
+  //Create the file that will log the messages received
+  int log_fd = open(LOGFILE_PATH, O_CREAT | O_RDWR | O_TRUNC, 0766);
+  if (log_fd == -1) {
+    syslog(LOG_ERR, "Error creating logfile =  %s. Exiting...", strerror(errno));
+    exit(-1);
+  }
+
+  int file_size = 0;
+  char *recv_data = NULL;
+  
+  while (graceful_exit_handler == false) {
+
+    struct sockaddr_storage client_addr;
+    socklen_t addr_size = sizeof client_addr;
+
+    int conn_fd = accept(socket_t, (struct sockaddr * ) & client_addr, & addr_size);
+    if (conn_fd == -1) {
+      if (errno == EINTR)
+        continue;
+      else {
+        syslog(LOG_ERR, "Error occured during accepting new connection = %s. Exiting...", strerror(errno));
+        exit(-1);
+      }
+    }
+    //Get IPv4 address of client
+    if (client_addr.ss_family == AF_INET) {
+      char addr[INET6_ADDRSTRLEN];
+      struct sockaddr_in * addr_in = (struct sockaddr_in * ) & client_addr;
+      inet_ntop(AF_INET, & (addr_in -> sin_addr), addr, INET_ADDRSTRLEN);
+      syslog(LOG_INFO, "Accepted connection from %s", addr);
+      printf("Accepted connection from %s\n", addr);
+    } else if (client_addr.ss_family == AF_INET6) {
+      char addr[INET6_ADDRSTRLEN];
+      struct sockaddr_in6 * addr_in6 = (struct sockaddr_in6 * ) & client_addr;
+      inet_ntop(AF_INET6, & (addr_in6 -> sin6_addr), addr, INET6_ADDRSTRLEN);
+      syslog(LOG_INFO, "Accepted connection from %s", addr);
+      printf("Accepted connection from %s\n", addr);
+    }
+
+    //Wait for data
+    int recv_ret;
+    //Start with a size, potentially incresing it if an entire packet cannot fit into it
+    recv_data = malloc(sizeof(char) * BUFFER_SIZE);
+    if(!recv_data){
+    	syslog(LOG_ERR, "Error occured during malloc = %s. Exiting...", strerror(errno));
+    	exit(-1);
+    }
+    int recv_idx = 0;
+    //How many 'BUFFER_SIZE' receive_blocks "recv_data" we have received from client till now
+    int receive_blocks = 1;
+    do {
+      recv_ret = recv(conn_fd, & recv_data[recv_idx], BUFFER_SIZE, 0);
+      if (recv_ret == -1) {
+        syslog(LOG_ERR, "Error occured while reading from socket = %s. Exiting...", strerror(errno));
+        exit(-1);
+      }
+//      printf("Received data = %s\n", &recv_data[recv_idx]);
+      recv_idx += recv_ret;
+
+      if (recv_idx != 0) {
+        if (recv_data[recv_idx - 1] == '\n') {
```

Comment:
@prvbharadwaj1731 what assumption are you making about the buffer returned by `recv` here?  What would happen if two packets were included in a single `recv` call?